src/game/moveFinding/findRookBaseMoves.js

```diff
@@ -1,3 +1,75 @@
-export default function findRookBaseMoves() {
-    return [];
+import BoardCoordinates from "../board/boardCoordinates";
+
+export default function findRookBaseMoves(boardSituation, startingCoordinates, color) {
+    const b = [...checkMovesVertical(boardSituation, startingCoordinates, color), 
+            ...checkMovesHorizontal(boardSituation, startingCoordinates, color)]; //makes one array of possible moves
+    return b;   //returns this array
+}
+
+
+function checkMovesHorizontal (boardSituation, startingCoordinates, pieceColor){
+    let x = startingCoordinates.x-1; //x position currently being checked
+    let movesHorizontal=[]; //array of possible moves
+    while(x>=0)  //check current row from piece position to left edge, stop upon finding another piece
+    {
+        let posValue = boardSituation[startingCoordinates.y][x];
```

Comment:
tu (i we wszystkich podobnych odniesieniach) powinno być raczej boardSituation.position[...][...] (oczywiście, można też wyciągnąć position na początku funkcji, żeby nie pisać w kółko odwołania do boardSituation).

W tej chwili to rozróżnienie nie jest oczywiste, bo na razie sprawdzamy tylko pozycję, ale sytuacja jedt obiektem zawierającym dodatkowe informacje poza aktualną pozycją - i pozycję trzeba dopiero z takiego obiektu wydobyć.

Reply:
W aktualnej wersji kodu funkcja przyjmuje zmienna boardSituation, ponieważ zakładam, że w przyszłości może to być potrzebne, a do nastepnych funkcji przekazuje zmienną, która przechowuje referencje na boardSituation.position